stregsystem/views.py

```diff
@@ -715,14 +713,23 @@ def api_quicksale(request, room, member: Member, bought_ids):
 
 def __append_bought_ids_to_product_list(products, bought_ids, time_now, room):
     try:
-        for i in bought_ids:
+        # Get the amount of unique items bought
+        unique_product_dict = {}
+        for unique_id in bought_ids:
+            if str(unique_id) not in unique_product_dict:
+                unique_product_dict[str(unique_id)] = 1
+            else:
+                unique_product_dict[str(unique_id)] += 1
```

Comment:
```suggestion
            if unique_id not in unique_product_dict:
                unique_product_dict[unique_id] = 1
            else:
                unique_product_dict[unique_id] += 1
```
what's the point in converting from an int object to string all the time?

Reply:
you can also optimize it further by just using dict comprehension:

```py
unique_product_dict = {i:bought_ids.count(i) for i in bought_ids}
```